cfme/fixtures/v2v.py

```diff
@@ -32,6 +28,29 @@ def host_creds(provider_setup):
     host.remove_credentials_rest()
 
 
+def _tag_cleanup(host_obj, tag1, tag2):
+    """
+        Clean Up Tags
+
+        Returns: Boolean True or False
+    """
+    tags = host_obj.get_tags()
+    if tags:
+        result = [
+            (object_tags.category.display_name == tag1.category.display_name and
```

Comment:
i took a second look at the expression

**required** extract the and simplify the shared nested expression

Reply:
i have a proposal here
```
# untested code example
def extract_tag(tag):
  return tag.category.display_name, tag2.display_name

valid_tags = {extract_tag(tag1), extract_tag(tag2)}
tags = host_obj.get_tags()
tags_set = set(map(extract_tag, tags))

if not tags_set.is_subset(valid_tags): # needs a check if i didn't mess up order ^^
   host_obj.remove_tags(tags=tags)
```

note that i didn't add a return value, since i didn't see it used

Reply:
@RonnyPfannschmidt, I really liked your idea but it is not possible here. 
Some of our tags have * in its category; so if you do get_tags(), it will fetch tags from UI details page which don't have * in there category name, so this is returning false while comparison.

For example:

```
In [80]: tag1 = appliance.collections.categories.instantiate(
        display_name='V2V - Transformation Host *').collections.tags.instantiate(
        display_name='t')

In [132]: tag1.category.display_name
Out[132]: 'V2V - Transformation Host *'

In [133]: host.get_tags().category.display_name
Out[133]: u'V2V - Transformation Host'
```

We can use something like partical_match here but I am not sure how should we compare with particial match (we usually do assignment with it, right?)

Rather than this, can we use this?

```
tags = vm.get_tags()

if tag1.display_name not in [item.display_name for item in tags]:
       host_obj.remove_tags(tags=tags)
```